### PR TITLE
minor internal cleanup in assert_close

### PR DESCRIPTION
**Pull request by pmeier**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #89559
* #90173
* #90172
* #90005
* #90542
* #90004
* __->__ #90003

Per title. I'm going to highlight them with inline comments.

### CODE REVIEW

File: torch/testing/_comparison.py [hunk 1]
```diff
@@ -333,25 +333,36 @@ def __init__(
         self.id = id
         self._unknown_parameters = unknown_parameters
 
+    @staticmethod
+    def _inputs_not_supported() -> NoReturn:
+        raise UnsupportedInputs()
```
Comment (pmeier) [inline torch/testing/_comparison.py:338]: Although, this looks quite unnecessary, it has one major advantage: raising `UnsupportedInputs` directly, leaks implementation details. For example, we no longer need to import `UnsupportedInputs` for `TestCase.assertEqual` and in there no one has to know what happens internally.

File: torch/testing/_comparison.py [hunk 2]
```diff
@@ -333,25 +333,36 @@ def __init__(
...
     @staticmethod
     def _check_inputs_isinstance(*inputs: Any, cls: Union[Type, Tuple[Type, ...]]):
         """Checks if all inputs are instances of a given class and raise :class:`UnsupportedInputs` otherwise."""
         if not all(isinstance(input, cls) for input in inputs):
-            raise UnsupportedInputs()
+            Pair._inputs_not_supported()
 
-    def _make_error_meta(self, type: Type[Exception], msg: str) -> ErrorMeta:
-        """Makes an :class:`ErrorMeta` from a given exception type and message and the stored id.
+    def _fail(
```
Comment (pmeier) [inline torch/testing/_comparison.py:346]: Two changes here:

1. For the same reason as above, this leaked details.
2. Just creating the exception but leaving the `raise` to the caller is fine in theory but one can easily botch this. For example, I myself forgot to `raise` it in a custom comparison in `torchvision`. It took us quite a while to even realize that we have unintentionally disabled part of the comparison and in turn we needed to untangle the mess from adding tests that should have failed but didn't. See pytorch/vision#6934.

File: torch/testing/_comparison.py [hunk 3]
```diff
@@ -617,9 +629,6 @@ class TensorLikePair(Pair):
             check is disabled, tensors with different ``layout``'s are converted to strided tensors before being
             compared.
         check_stride (bool): If ``True`` and corresponding tensors are strided, asserts that they have the same stride.
-        check_is_coalesced (bool): If ``True`` (default) and corresponding tensors are sparse COO, checks that both
```
Comment (pmeier) [inline torch/testing/_comparison.py:620]: We removed that in #68728, but forgot to update the docstring.

**Review by mruberry — APPROVED** [review]
OK

**Review by malfet — APPROVED** [review]

**Pull request closed without merging**